### PR TITLE
[SYSTEMML-446] Bug fixes to SYSTEMML-446

### PR DESCRIPTION
**Pull request by nakul02**

### CODE REVIEW

**Comment by nakul02** [comment]
@niketanpansare - Could you please take a look at these?

File: src/main/java/org/apache/sysml/runtime/instructions/cp/ReorgCPInstruction.java [hunk 1]
```diff
@@ -130,6 +130,9 @@ public void processInstruction(ExecutionContext ec)
 		//execute operation
 		MatrixBlock soresBlock = (MatrixBlock) (matBlock.reorgOperations(r_op, new MatrixBlock(), 0, 0, 0));
         
+        // After a reorg operation, it may be eligible to become sparse
+	    soresBlock.recomputeNonZeros();
```
Comment (niketanpansare) [inline src/main/java/org/apache/sysml/runtime/instructions/cp/ReorgCPInstruction.java]: This is extremely surprising. Are we sure about this ?

**Comment by niketanpansare** [comment]
LGTM. Thanks Nakul.

@deroneriksson @dusenberrymw Can you please merge this in ?

**Comment by deroneriksson** [comment]
I will merge. Thank you @nakul02.

**Comment by nakul02** [comment]
Thanks @deroneriksson

**Pull request closed without merging**